packages/core/src/config/configNormalizers/stackTrace.js

```diff
@@ -0,0 +1,161 @@
+  if (!config) {
+    config = {};
+  }
+
+  if (!config.tracing) {
+    config.tracing = {};
+  }
+
+  const stackTrace = normalizeStackTraceMode(config.tracing);
+  const stackTraceLength = normalizeStackTraceLength(config.tracing);
+
+  return {
+    stackTrace,
+    stackTraceLength
+  };
+};
+
+/**
+ * Normalizes stack trace configuration from agent.
+ *
+ * @param {*} config - tracing.global config from agent containing stack-trace and stack-trace-length
+ * @returns {{ stackTrace: string | null, stackTraceLength: number | null }} - Normalized values
+ */
+exports.normalizeAgentConfig = function normalizeAgentConfig(config) {
+  if (!config || typeof config !== 'object') {
+    return { stackTrace: null, stackTraceLength: null };
+  }
+
+  const tracingObj = {
+    stackTrace: config['stack-trace'],
+    stackTraceLength: config['stack-trace-length']
+  };
+
+  const result = {
+    stackTrace: normalizeStackTraceMode(tracingObj, true),
+    stackTraceLength: normalizeStackTraceLength(tracingObj, true)
+  };
+
+  return result;
+};
+
+/**
+ * Normalizes stack trace mode configuration.
+ *
+ * @param {*} tracingObj - The tracing object containing stackTrace
+ * @param {boolean} [isAgentConfig=false] - Whether this is from agent config
+ * @returns {string | null} - Normalized value or null if from agent and invalid
+ */
+function normalizeStackTraceMode(tracingObj, isAgentConfig = false) {
+  const envVar = 'INSTANA_STACK_TRACE';
+
+  if (isAgentConfig) {
+    const configPath = 'stack-trace value from agent';
+
+    const validatedAgentConfig = validateStackTraceMode(tracingObj?.stackTrace, configPath);
+
+    if (validatedAgentConfig != null) {
+      return validatedAgentConfig;
+    }
+
+    return null;
+  }
+
+  const validatedConfigGlobal = validateStackTraceMode(
+    tracingObj?.global?.stackTrace,
+    'config.tracing.global.stackTrace'
+  );
+  if (validatedConfigGlobal != null) {
+    return validatedConfigGlobal;
+  }
+
+  const validatedConfig = validateStackTraceMode(tracingObj?.stackTrace, 'config.tracing.stackTrace');
+  if (validatedConfig != null) {
+    return validatedConfig;
+  }
+
+  if (process.env[envVar]) {
+    const validatedEnv = validateStackTraceMode(process.env[envVar], envVar);
+    if (validatedEnv != null) {
+      return validatedEnv;
+    }
+  }
+
+  return DEFAULT_STACK_TRACE_MODE;
+}
+
+/**
+ * Normalizes stack trace length configuration.
+ *
+ * @param {*} tracingObj - The tracing object containing stackTraceLength
+ * @param {boolean} [isAgentConfig=false] - Whether this is from agent config
+ * @returns {number | null} - Normalized value or null if from agent and invalid
+ */
+function normalizeStackTraceLength(tracingObj, isAgentConfig = false) {
+  const envVar = 'INSTANA_STACK_TRACE_LENGTH';
+
+  if (isAgentConfig) {
+    const configPath = 'stack-trace-length value from agent';
+
+    const validatedAgentConfig = validateStackTraceLength(tracingObj?.stackTraceLength, configPath);
```

Comment:
I feel like we still mix validation and normalization :D
The parent caller should deal with that - otherwise you are coupling the code too much